ruby.sh: line 19: bad substitution: no closing `)' in <(

# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [x] your problem was from running the official `install` or `uninstall` script?
- [x] after installation: ran `brew config` and `brew doctor` and included their output with your issue? If you couldn't install: provided your OS version with the output of your issue?

<!-- To help us debug your issue, please complete these sections: -->

## What you were trying to do (and why)

Install Homebrew

## What happened (include command output)

Saw a warning/error:
```
~/.linuxbrew/Homebrew/Library/Homebrew/utils/ruby.sh: line 19: bad substitution: no closing `)' in <(
```
Expand below for details.

<details>
  <summary>Command output</summary>
  <pre>
…
==> Tapping homebrew/core
remote: Enumerating objects: 1097704, done.
remote: Counting objects: 100% (75/75), done.
remote: Compressing objects: 100% (35/35), done.
remote: Total 1097704 (delta 40), reused 75 (delta 40), pack-reused 1097629
Receiving objects: 100% (1097704/1097704), 418.71 MiB | 30.24 MiB/s, done.
Resolving deltas: 100% (756062/756062), done.
From https://github.com/Homebrew/homebrew-core
 * [new branch]              master     -> origin/master
Updating files: 100% (6223/6223), done.
HEAD is now at ccd98dd9e75 repo: update 2.18 bottle.
/mnt/home/shaun.jackman/.linuxbrew/Homebrew/Library/Homebrew/utils/ruby.sh: line 19: bad substitution: no closing `)' in <(
      # function which() is set by brew.sh
      # it's aliased to `type -P`
      # shellcheck disable=SC2230
      which -a ruby
      # shellcheck disable=SC2230
      PATH="${HOMEBREW_PATH}" which -a ruby
    )
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
Already downloaded: /mnt/home/shaun.jackman/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
==> Pouring portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
==> Installation successful!
…
  </pre>
</details>

## What you expected to happen

No warning/error
```
~/.linuxbrew/Homebrew/Library/Homebrew/utils/ruby.sh: line 19: bad substitution: no closing `)' in <(
      # function which() is set by brew.sh
      # it's aliased to `type -P`
      # shellcheck disable=SC2230
      which -a ruby
      # shellcheck disable=SC2230
      PATH="${HOMEBREW_PATH}" which -a ruby
    )
```

## Step-by-step reproduction instructions (by running `brew` commands)

```console
$ /bin/bash --version | head -n1
GNU bash, version 4.2.46(2)-release (x86_64-koji-linux-gnu)
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

## `brew config`

```
HOMEBREW_VERSION: 3.3.5-25-g7c4247e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c4247eec9685071cbadc52952d0f2afadf022cc
Last commit: 7 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ccd98dd9e75bd87e58b919cbe1d35628ce1398b8
Core tap last commit: 44 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /mnt/home/shaun.jackman/.linuxbrew
HOMEBREW_REPOSITORY: /mnt/home/shaun.jackman/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /mnt/home/shaun.jackman/.linuxbrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DEVELOPER: set
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 96
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
Homebrew Ruby: 2.6.8 => /mnt/home/shaun.jackman/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 96-core 64-bit skylake
Clang: N/A
Git: 2.26.1 => /bin/git
Curl: 7.76.1 => /bin/curl
Kernel: Linux 5.4.74-36.135.amzn2.x86_64 x86_64 GNU/Linux
OS: Amazon Linux release 2 (Karoo) (Karoo)
Host glibc: 2.26
/usr/bin/gcc: 7.3.1
/usr/bin/ruby: 2.0.0
glibc: N/A
gcc@5: N/A
xorg: N/A
```

## `brew doctor`

```
Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.
```

--- a/Library/Homebrew/utils/ruby.sh
+++ b/Library/Homebrew/utils/ruby.sh
@@ -31,7 +31,7 @@ find_ruby() {
       fi
     done < <(
       # function which() is set by brew.sh
-      # it's aliased to `type -P`
+      # it is aliased to `type -P`
       # shellcheck disable=SC2230
       which -a ruby
       # shellcheck disable=SC2230

